modules/creating-manifest-file-customized-br-ex-bridge.adoc

```diff
@@ -0,0 +1,257 @@
+. Create an `extraworker-secret` `Secret` object that has a minimal static IP configuration.
+
+. Apply the `extraworker-secret` `Secret` object to each node in your cluster by entering the following command. This step provides each compute node access to the Ignition config file.
++
+[source,terminal]
+----
+$ oc apply -f ./extraworker-secret.yaml
+----
+
+. Create a `BareMetalHost` resource and specify the network secret in the `preprovisioningNetworkDataName` parameter.
++
+.Example `BareMetalHost` resource with an attached network secret
+[source,yaml]
+----
+apiVersion: metal3.io/v1alpha1
+kind: BareMetalHost
+spec:
+# ...
+  preprovisioningNetworkDataName: ostest-extraworker-0-network-config-secret
+# ...
+----
+
+. To mange the `BareMetalHost` object within the `openshift-machine-api` namespace of your cluster, enter the following command:
++
+[source,terminal]
+----
+$ oc project openshift-machine-api
+----
+
+. Get the machine sets:
++
+[source,terminal]
+----
+$ oc get machinesets
+----
+
+. Scale each machine set by entering the following command. You must run this command for each machine set.
++
+[source,terminal]
+----
+$ . oc scale machineset <machineset_name> --replicas=<n> <1>
+----
+<1> Where `<machineset_name>` is the name of the machine set and `<n>` is the number of compute nodes.
++
+Each compute node automatically reboots to receive the network configuration specified in `preprovisioningNetworkDataName` and the configuration in the `MachineConfig` CRD.
+
+
+endif::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+ifdef::postinstall-bare-metal-ipi,postinstall-bare-metal-upi[]
+. Create a `NodeNetworkConfigurationPolicy (NNCP)` manifest file and define a customized `br-ex` bridge network configuration. Depending on your needs, ensure that you set a masquerade IP for either the `ipv4.address.ip`, `ipv6.address.ip`, or both parameters. A masquerade IP address must match an in-use IP address block.
++
+.Example of an NNCP manifest that sets IPv6 and IPv4 masquerade IP addresses.
+[source,yaml]
+----
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: worker-0-br-ex <1>
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: worker-0
+    desiredState:
+    interfaces:
+    - name: enp2s0 <2>
+      type: ethernet <3>
+      state: up <4>
+      ipv4:
+        enabled: false <5>
+      ipv6:
+        enabled: false
+    - name: br-ex
+      type: ovs-bridge
+      state: up
+      ipv4:
+        enabled: false
+        dhcp: false
+      ipv6:
+        enabled: false
+        dhcp: false
+      bridge:
+        port:
+        - name: enp2s0 <6>
+        - name: br-ex
+    - name: br-ex
+      type: ovs-interface
+      state: up
+      copy-mac-from: enp2s0
+      ipv4:
+        enabled: true
+        dhcp: true
+        address:
+        - ip: "169.254.169.2"
+          prefix-length: 29
+      ipv6:
+        enabled: false
+        dhcp: false
+        address:
+        - ip: "fd69::2/128"
```

Comment:
Yes, this needs a prefix length. I also checked on the exact value for this address and I had it slightly wrong. This should be ip="fd69::2", prefix-length="125", not 128 as I had originally written.

Reply:
:heavy_check_mark: